system/src/system_sleep.cpp

```diff
@@ -116,6 +116,12 @@ int system_sleep_impl(Spark_Sleep_TypeDef sleepMode, long seconds, uint32_t para
     //---- #2
     // SLEEP_NETWORK_STANDBY can keep the modem on during DEEP sleep
     // System.sleep(10) always powers down the network, even if SLEEP_NETWORK_STANDBY flag is used.
+
+    // Make sure all confirmable UDP messages are sent and acknowledged before sleeping
+    if (spark_cloud_flag_connected() && !(param & SYSTEM_SLEEP_FLAG_NO_WAIT)) {
+        Spark_Sleep();
```

Comment:
This is a blocking call. Do we want to keep it like that?

Reply:
For now I'd say yes, because `System.sleep()` is a blocking call as well. Also, we're already waiting for confirmable messages synchronously when entering the stop mode: https://github.com/particle-iot/device-os/blob/5068605f4fb4d80bfc6ba366d1a5da61db60480f/system/src/system_sleep.cpp#L164